homeassistant/components/texecom.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for Texecom Alarm Panels & Devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/texecom/
+"""
+
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import (async_dispatcher_send)
+
+REQUIREMENTS = ['pyTexecom==0.2.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'texecom'
+
+DATA_EVL = 'texecom'
+
+CONF_PORT = 'port'
+CONF_PANEL_TYPE = 'panel_type'
+CONF_ZONE_NAME = "name"
+CONF_ZONES = 'zones'
+CONF_ZONE_TYPE = 'type'
+CONF_ZONE_NUMBER = 'zone_number'
+
+DEFAULT_PORT = '/dev/ttys0'
+DEFAULT_ZONE_NAME = 'zone'
+DEFAULT_ZONE_NUMBER = '1'
+DEFAULT_ZONE_TYPE = 'motion'
+
+SIGNAL_ZONE_UPDATE = 'texecom.zones_updated'
+
+ZONE_SCHEMA = vol.Schema({
+    vol.Required(CONF_ZONE_NAME): cv.string,
+    vol.Required(CONF_ZONE_TYPE, default=DEFAULT_ZONE_TYPE): cv.string,
+    vol.Required(CONF_ZONE_NUMBER, default=DEFAULT_ZONE_NUMBER): cv.string,
```

Comment:
Why not move the zone number value to the outer dict key so that it's logical why we need the integer key in the outer dict? I assume that's why it's needed, ie that we shouldn't define zones with the same number.

```py
ZONE_SCHEMA = vol.Schema({
    vol.Required(CONF_ZONE_NAME): cv.string,
    vol.Required(CONF_ZONE_TYPE, default=DEFAULT_ZONE_TYPE): cv.string,
})
```